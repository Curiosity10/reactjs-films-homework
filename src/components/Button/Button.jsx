import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import styles from './Button.scss';

const Button = (props) => {
  const { children, className, handleClick } = props;
  return (
    <button
      aria-label={children}
      type="button"
      onClick={handleClick}
      className={cn(styles.btn, styles[className])}
    >
      { children }
    </button>
  );
};

Button.propTypes = {
  className: PropTypes.string.isRequired,
  handleClick: PropTypes.func,
  children: PropTypes.string.isRequired,
};

Button.defaultProps = {
  handleClick: () => {},
};

export default Button;
