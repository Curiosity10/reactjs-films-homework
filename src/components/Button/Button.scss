@import "../../styles/variables.scss";

.btn {
  cursor: pointer;
  color: white;
  border-radius: 3px;
  width: 100px;
  max-height: 30px;
  padding: 5px;
  font-family: "Helvetica Neue", Arial, Helvetica, sans-serif;
  outline: none;
}

.transparentBtn {
  border: 1px solid white;
  background-color: transparent;

  &:hover {
    transform: scale(1.05);
  }
}

.blueBtn {
  border: none;
  background-color: $blue-color;

  &:hover {
    transform: scale(1.05);
  }
}
