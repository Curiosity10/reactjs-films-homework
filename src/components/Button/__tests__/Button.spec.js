import React from 'react';
import Button from '..';
import { create, act } from 'react-test-renderer';

describe('Button component renders correctly', () => {
  it('Blue button renders correctly', () => {
    const tree = create(<Button className="blueBtn">Watch Now</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Transparent button renders correctly', () => {
    const tree = create(<Button className="transparentBtn">View Info</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Click button with handler works correctly', () => {
    const mockFunction = jest.fn();
    const tree = create(
      <Button
        handleClick={mockFunction}
        className="transparentBtn"
      >
        View Info
      </Button>,
    );
    const button = tree.root.children[0];
    act(() => {
      button.props.onClick();
    });
    expect(mockFunction.mock.calls.length).toEqual(1);
  });

  it('Click button without handler works correctly', () => {
    const tree = create(
      <Button
        className="transparentBtn"
      >
        View Info
      </Button>,
    );
    const button = tree.root.children[0];
    const mockFunction = jest.fn(button.props.onClick);
    act(() => {
      mockFunction();
    });
    expect(mockFunction).toHaveBeenCalled();
  });
});
