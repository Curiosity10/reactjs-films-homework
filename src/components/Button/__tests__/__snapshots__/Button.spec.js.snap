// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly Blue button renders correctly 1`] = `
<button
  aria-label="Watch Now"
  className="btn blueBtn"
  onClick={[Function]}
  type="button"
>
  Watch Now
</button>
`;

exports[`Button component renders correctly Transparent button renders correctly 1`] = `
<button
  aria-label="View Info"
  className="btn transparentBtn"
  onClick={[Function]}
  type="button"
>
  View Info
</button>
`;
